experimental/devices/mfrc522/example_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package mfrc522_test
+
+import (
+	"encoding/hex"
+	"log"
+	"reflect"
+	"time"
+
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/experimental/devices/mfrc522"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+	"periph.io/x/periph/host"
+	"periph.io/x/periph/host/rpi"
+)
+
+func Example() {
+	// Make sure periph is initialized.
+	if _, err := host.Init(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Using SPI as an example. See package ./spi/spireg for more details.
+	p, err := spireg.Open("")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer p.Close()
+
+	rfid, err := mfrc522.NewSPI(p, rpi.P1_13, rpi.P1_11)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Idling device on exit.
+	defer rfid.Halt()
+
+	// Setting the antenna signal strength.
+	rfid.SetAntennaGain(5)
+
+	// Converting access key.
+	// This value corresponds to first pi "numbers": 3 14 15 92 65 35.
+	hexKey, _ := hex.DecodeString("030e0f5c4123")
+	var key [6]byte
+	copy(key[:], hexKey)
+
+	// Converting expected data.
+	// This value corresponds to string "@~>f=Um[X{LRwA3}".
+	expected, _ := hex.DecodeString("407e3e663d556d5b587b4c527741337d")
+
+	timedOut := false
+	cb := make(chan []byte)
+	timer := time.NewTimer(10 * time.Second)
+
+	// Stopping timer, flagging reader thread as timed out
+	defer func() {
+		timer.Stop()
+		timedOut = true
+		close(cb)
+	}()
+
+	go func() {
+		log.Printf("Started %s", rfid.String())
+
+		for {
+			// Trying to read data from sector 1 block 0
+			data, err := rfid.ReadCard(byte(commands.PICC_AUTHENT1B), 1, 0, key)
+
+			// If main thread timed out just exiting.
+			if timedOut {
+				return
+			}
+
+			// Some devices tend to send wrong data while RFID chip is already detected
+			// but still "too far" from a receiver.
+			// Especially some cheap CN clones which you can find on GearBest, AliExpress, etc.
+			// This will suppress such errors.
```

Comment:
What about having a clear distinction in the error returned by ReadCard() to distinguish between bus error and logical error? Is it possible to distinguish?

Reply:
Unfortunately, you can get a whole variety of errors. I have devices from 3 different manufacturers and those two which are from China are sending different signals every time when a card is on the edge of antenna radius :(

Reply:
I specifically mean that I would expect in this case the error to be one of:
- AuthReadFailure / "can not authenticate"
- In LowLevel.Auth() differentiate when `n&0x08 == 0` triggers vs DevRead() returns an error.
- "expected 16 bytes, actual %d"
- "can't read data after 2000 loops"
- "IRQ error"

vs sysfs SPI driver reporting an error.